src/formElementPlugins/listBuilder/src/ApplicationView.tsx

```diff
@@ -0,0 +1,347 @@
+import React, { useEffect, useState } from 'react'
+import {
+  Button,
+  Modal,
+  Form,
+  Segment,
+  Table,
+  TableHeaderCell,
+  TableCell,
+  Card,
+  Icon,
+  Label,
+} from 'semantic-ui-react'
+import { ApplicationViewProps } from '../../types'
+import { ResponseFull } from '../../../utils/types'
+import { TemplateElement, TemplateElementCategory } from '../../../utils/generated/graphql'
+import ApplicationViewWrapper from '../../ApplicationViewWrapper'
+import strings from '../constants'
+
+enum DisplayType {
+  CARDS = 'cards',
+  TABLE = 'table',
+}
+
+interface InputResponseField {
+  id?: number
+  code?: string
+  isValid?: boolean
+  title?: string | null | undefined
+  value: ResponseFull
+}
+
+type ListItem = { [code: string]: InputResponseField }
+
+const ApplicationView: React.FC<ApplicationViewProps> = ({
+  element,
+  parameters,
+  onSave,
+  onUpdate,
+  Markdown,
+  initialValue,
+  currentResponse,
+  applicationData,
+  allResponses,
+}) => {
+  const { isEditable } = element
+  const {
+    label,
+    description,
+    createModalButtonText = strings.BUTTON_LAUNCH_MODAL,
+    modalText,
+    addButtonText = strings.BUTTON_ADD,
+    updateButtonText = strings.BUTTON_UPDATE,
+    deleteItemText = strings.BUTTON_DELETE,
+    inputFields,
+    displayFormat = getDefaultDisplayFormat(inputFields),
+    displayType = DisplayType.CARDS,
+  } = parameters
+
+  const [currentInputResponses, setCurrentInputResponses] = useState<InputResponseField[]>(
+    resetCurrentResponses(inputFields)
+  )
+
+  const [listItems, setListItems] = useState<ListItem[]>(initialValue?.list ?? [])
+  const [selectedListItem, setSelectedListItem] = useState<number | null>(null)
+  const [open, setOpen] = useState(false)
+  const [inputError, setInputError] = useState(false)
+
+  useEffect(() => {
+    onSave({
+      text: 'Not required', // Text value never displayed to user
+      list: listItems,
+    })
+  }, [listItems])
+
+  // This is sent to ApplicationViewWrapper and runs instead of the default responseMutation
+  const innerElementUpdate = async ({ variables: response }: { variables: InputResponseField }) => {
+    const newResponses = [...currentInputResponses]
+    newResponses[response.id as number] = response
+    setCurrentInputResponses(newResponses)
+    if (!anyErrorItems(newResponses, inputFields)) setInputError(false)
```

Comment:
should this only happen on submit add/update button ? (an setInputError(false) after adding or updating ?)

Reply:
The main `setInputError` happens on Submit/update. But this is an extra one so that if you have the error message showing on the modal, it will disappear as soon as you leave the last invalid field. (i.e. you don't want to the message to persist once you've fixed all the invalid fields)

Reply:
ahh yep, sounds good